src/js/Autodiscover/AutodiscoverService.ts

```diff
@@ -512,18 +512,18 @@ export class AutodiscoverService extends ExchangeServiceBase {
 
         var host = hosts[currentHostIndex];
         // var isScpHost:bool = currentHostIndex < scpHostCount;
-        var autodiscoverUrlOut:IOutParam<Uri> = { outValue:null };
+        var autodiscoverUrlOut: IOutParam<Uri> = { outValue: null };
         return this.TryGetAutodiscoverEndpointUrl(host, autodiscoverUrlOut)
             .then<TGetSettingsResponseCollection>((value) => {
                 if (value) {
+                    // If we got this far, the response was successful, set Url.
+                    this.Url = autodiscoverUrlOut.outValue;
```

Comment:
Note: This is independent from the NTLM thing, it just seems to be wrong in general? I think the existing line does not make sense because autodiscoverUrlRef seems to be pointing to this.Url as well, so nothing is achieved at all...